.gitignore

```diff
@@ -1,14 +1,124 @@
-.idea/*
-.vscode/ipch
-build
-builder/output
-builder/deployed
-builder/check_aesni
-builder/credentials
-db
-*.zip
-*.patch
+.DS_Store
+/doc
+/build
+/tags
+log/
+
+# vim swap files
+*.swo
+TAGS
+!TAGS/
+tags
+!tags/
+gtags.files
+GTAGS
+GRTAGS
+GPATH
+cscope.files
+cscope.out
+cscope.in.out
+cscope.po.out
+
 external/miniupnpc/Makefile
 miniupnpcstrings.h
 version/
-cmake-build-debug/*
+ClangBuildAnalyzerSession.txt
+
+# gitian
+.venv
+contrib/gitian/builder/
+contrib/gitian/docker/
+contrib/gitian/sigs/
+
+# Created by https://www.gitignore.io
+
+### C++ ###
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Fortran module files
+*.mod
+
+# Compiled Static libraries
+*.lai
+*.la
+*.a
+*.lib
+
+# Executables
+*.exe
+*.out
+*.app
+
+
+### CMake ###
+CMakeCache.txt
+CMakeFiles
+cmake_install.cmake
+install_manifest.txt
+cmake-build-debug/
+
+### Linux ###
+*~
+
+# KDE directory preferences
+.directory
+
+### VSCode ###
+.vscode/
+
+### CLion ###
+.idea/
+
+### Eclipse ###
+*.pydevproject
+.metadata
+.gradle
+bin/
+tmp/
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
+.settings/
+.loadpath
+
+# External tool builders
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# CDT-specific
+.cproject
+
+# PDT-specific
+.buildpath
+
+# Netbeans-specific
+nbproject
```

Comment:
```suggestion
```

we have no java stuffs here

Reply:
Yeah probably because previous dev used Eclipse and left those in here so that the IDE settings are not committed to VCS.